plugins/modules/terraform.py

```diff
@@ -312,10 +312,10 @@ def preflight_validation(bin_path, project_path, version, variables_args=None, p
 
 
 def _state_args(state_file):
```

Comment:
The function does the right thing, so just as a matter of style, I would suggest:
```suggestion
def _state_args(state_file):
    if not state_file:
        return []
    if not os.path.exists(state_file):
        module.warn(msg='Could not find state_file "{0}", the process will not destroy any resources, please check your state file path.'.format(state_file))
    return ['-state', state_file]
```